src/storage-client/src/types/connections.rs

```diff
@@ -372,6 +404,36 @@ impl KafkaConnection {
 
         Ok(config.create_with_context(context)?)
     }
+
+    async fn validate(&self, connection_context: &ConnectionContext) -> Result<(), anyhow::Error> {
+        let consumer: BaseConsumer<_> = self
+            .create_with_context(connection_context, MzClientContext, &BTreeMap::new())
+            .await?;
+
+        // librdkafka doesn't expose an API for determining whether a connection to
+        // the Kafka cluster has been successfully established. So we make a
+        // metadata request, though we don't care about the results, so that we can
+        // report any errors making that request. If the request succeeds, we know
+        // we were able to contact at least one broker, and that's a good proxy for
+        // being able to contact all the brokers in the cluster.
+        //
+        // The downside of this approach is it produces a generic error message like
+        // "metadata fetch error" with no additional details. The real networking
+        // error is buried in the librdkafka logs, which are not visible to users.
+        //
+        // TODO(benesch): pull out more error details from the librdkafka logs and
+        // include them in the error message.
```

Comment:
Can we structure the code to avoid copy/pasting this whole comment? Maybe a shared `validate` method in the `kafka-util` crate?

Reply:
Oops, I had already done this and forgot to delete kafka-util version which was dead code. Deleted!